include/matx/operators/interp.h

```diff
@@ -428,105 +428,47 @@ namespace matx {
+          cudaStream_t stream = ex.getStream();
 
-          index_t _batch_count = 1;
+          index_t batch_count = 1;
           for (int i = 0; i < v_.Rank() - 1; i++) {
-            _batch_count *= v_.Size(i);
-          }
-          index_t _n = v_.Size(v_.Rank() - 1);
-          if (_batch_count > std::numeric_limits<int>::max() || _n > std::numeric_limits<int>::max()) {
-            const std::string err_msg = "Spline interpolation is not supported for tensors with more than 2^" + std::to_string(std::numeric_limits<int>::digits) + " items";
-            MATX_THROW(matxInvalidSize, err_msg.c_str());
+            batch_count *= v_.Size(i);
           }
-          int batch_count = static_cast<int>(_batch_count);
-          int n = static_cast<int>(_n); // number of sample points
+          index_t n = v_.Size(v_.Rank() - 1);
 
+          
           cuda::std::array m_shape = v_.Shape();
           detail::AllocateTempTensor(m_, std::forward<Executor>(ex), m_shape, &ptr_m_);
 
-          detail::tensor_impl_t<value_type, OpV::Rank()> d_tensor, dl_tensor, du_tensor; // Derivatives at sample points (spline only)
-          value_type *ptr_dl_ = nullptr;
-          value_type *ptr_d_ = nullptr;
-          value_type *ptr_du_ = nullptr;
+          // Allocate temporary storage for tridiagonal system
+          // use a single buffer for all three diagonals so that we can use the DIA format
+          value_type *ptr_tridiag_ = nullptr;
+          matxAlloc((void**)&ptr_tridiag_, 3 * batch_count * n * sizeof(value_type), MATX_MANAGED_MEMORY, stream);
+          value_type *ptr_dl_ = ptr_tridiag_;
+          value_type *ptr_d_  = ptr_tridiag_ + batch_count * n;
+          value_type *ptr_du_ = ptr_tridiag_ + batch_count * n * 2;
 
-          detail::AllocateTempTensor(dl_tensor, std::forward<Executor>(ex), m_shape, &ptr_dl_);
-          detail::AllocateTempTensor(d_tensor, std::forward<Executor>(ex), m_shape, &ptr_d_);
-          detail::AllocateTempTensor(du_tensor, std::forward<Executor>(ex), m_shape, &ptr_du_);
+          detail::tensor_impl_t<value_type, OpV::Rank()> dl_tensor, d_tensor, du_tensor; // Derivatives at sample points (spline only)
+          make_tensor(dl_tensor, ptr_dl_, m_shape);
+          make_tensor(d_tensor, ptr_d_, m_shape);
+          make_tensor(du_tensor, ptr_du_, m_shape);
 
           // Fill tridiagonal system via custom operator
-          InterpSplineTridiagonalFillOp(dl_tensor,d_tensor, du_tensor, m_, x_, v_).run(std::forward<Executor>(ex));
+          InterpSplineTridiagonalFillOp(dl_tensor, d_tensor, du_tensor, m_, x_, v_).run(std::forward<Executor>(ex));
+
+          // // Convert to uniform batched dia format
+          auto val_tensor = make_tensor(ptr_tridiag_, {batch_count * n * 3});
+          auto offset_tensor = make_tensor<index_t>({3}, MATX_MANAGED_MEMORY, stream);
+          offset_tensor.SetVals({-1, 0, 1});
```

Comment:
The PR with

make_tensor_uniform_batched_tri_dia(ValTensor &val,
                                         const index_t (&shape)[3])

has been submitted, so you no longer have to define the offsets yourself!